frontend/src/templates.ts

```diff
@@ -39,3 +51,20 @@ export const templates = {
     `;
   },
 };
+
+/**
+ * Transform a hierarchy object into a displayable string
+ *
+ * @param {Hierarchy} hierarchy A record's hierarchy under the form { lvl0: '', lvl1: '', lvl2: '', ... }
+ * @returns {string} A string representation of the hierarchy,
```

Comment:
Why does it limit to lvl1 to 3 ?

Reply:
lvl3 means `<h4>`, I don't have any example with `<h5>` (I don't even have pages with `<h4>` actually I think) and I'm afraid it would bring more noise than value.

Reply:
then the extractor in the backend should be changed, we currently index that `h1, h2, h3, h4, h5, h6` but if we never use it it's no use.

Reply:
also remember that we can have `h1, null, null, null, h5` 
maybe we can make something smarter and displays `> ... > ` in the middle if they are more than N headings

Reply:
> but if we never use it it's no use

It allows people that want to build their own UI to have everything they could need. That's why we extract many other fields, even if our UI doesn't use them: `image`, `lang`, `authors`, `date`...

> also remember that we can have h1, null, null, null, h5

Ok I can keep the 3 first headings

> maybe we can make something smarter and displays `> ... >` in the middle if they are more than N headings

I think it's more important for the context to keep the heavier headings.